File: algorithms/cpp/392.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int m = s.size(), n = t.size();
        int i = 0, j = 0;
        while (i < n) {
            if (j < m && t[i] == s[j])
                j += 1;
            i += 1;
        }
        return j == m;
    }
};

int main() {
    return 0;
}

File: algorithms/rust/392.rs
impl Solution {
    #[allow(unused)]
    pub fn is_subsequence(s: String, t: String) -> bool {
        let mut t = t.chars();
        for c in s.chars() {
            loop {
                match t.next() {
                    Some(ch) => {
                        if c == ch {
                            break;
                        }
                    }
                    _ => return false,
                }
            }
        }
        true
    }
}

#[allow(dead_code)]
struct Solution;

fn main() {
    println!("Hello, world!");
}
